scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/example/nnframes/xgboost/xgbClassifierTrainingExampleOnCriteoClickLogsDataset.scala

```diff
@@ -0,0 +1,110 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.dllib.examples.nnframes.xgboost
+
+import ml.dmlc.xgboost4j.scala.spark.TrackerConf
+import com.intel.analytics.bigdl.dllib.NNContext
+import com.intel.analytics.bigdl.dllib.nnframes.XGBClassifier
+import org.apache.spark.ml.feature.{StringIndexer, VectorAssembler}
+import org.apache.spark.sql.types.{StructField, StructType, LongType}
+import org.apache.spark.sql.{SQLContext, SparkSession, Row}
+import org.apache.spark.SparkContext
+
+class Task extends Serializable{
+  def rowToLibsvm(row: Row): String = {
+    0 until row.length flatMap {
+      case 0 => Some(row(0).toString)
+      case i if row(i) == null => Some("-999")
+      case i => Some( (if (i < 14) row(i) else java.lang.Long.parseLong(row(i).toString, 16)).toString )
+    } mkString " "
+  }
+}
+
+object xgbClassifierTrainingExampleOnCriteoClickLogsDataset {
+  def main(args: Array[String]): Unit = {
+    if (args.length < 3) {
+      println("Usage: program input_path modelsave_path num_threads")
+      sys.exit(1)
+    }
+
+    val sc = NNContext.initNNContext()
+    val spark = SQLContext.getOrCreate(sc)
+    val task = new Task()
+
+    val input_path = args(0) // path to data
+    val modelsave_path = args(1) // save model to this path
+    val num_threads = args(2).toInt // xgboost threads
+
+    var df = spark.read.option("header", "false").option("inferSchema", "true").option("delimiter", "\t").csv(input_path)
+    val processedRdd = df.rdd.map(task.rowToLibsvm)
+
+    var structFieldArray = new Array[StructField](40)
+    for(i <- 0 to 39) {
+      structFieldArray(i) = StructField("_c" + i.toString, LongType, true)
+    }
+    var schema =  new StructType(structFieldArray)
+
+    val rowRDD = processedRdd.map(_.split(" ")).map(row => Row.fromSeq(
+      for{
+        i <- 0 to 39
+      } yield {
+        row(i).toLong
+      }
+    ))
+
+    df = spark.createDataFrame(rowRDD,schema)
+
+    val stringIndexer = new StringIndexer()
+      .setInputCol("_c0")
+      .setOutputCol("classIndex")
+      .fit(df)
+    val labelTransformed = stringIndexer.transform(df).drop("_c0")
+
+    var inputCols = new Array[String](39)
+    for(i <- 0 to 38){
+      inputCols(i) = "_c" + (i+1).toString
+    }
+
+    val vectorAssembler = new VectorAssembler().
+      setInputCols(inputCols).
+      setOutputCol("features")
+
+    val xgbInput = vectorAssembler.transform(labelTransformed).select("features","classIndex")
+    val Array(train, eval1, eval2, test) = xgbInput.randomSplit(Array(0.6, 0.2, 0.1, 0.1))
+
+    val xgbParam = Map("tracker_conf" -> TrackerConf(0L, "scala"),
+      "eval_sets" -> Map("eval1" -> eval1, "eval2" -> eval2),
+    )
+
```

Comment:
All stage before XGBoost is Pre-processing stage. Need add comment about stage. Such that other developer can understand better.